mullvad/mullvadvpn-app
# Test updating between app versions and nightly test runs

This PR adds tests for verifying that settings are successfully migrated when updating from an older app version to the current app version. Also nightly test runs have been configured.

For running settings migration tests there's a new workflow iOS settings migration tests(`ios-end-to-end-tests-settings-migration.yml`). It installs an old app version, runs test changing settings, installs current app version and runs a test verifying that the settings are still changed. Since custom DNS cannot be set when DNS blockers are activated there has to be two runs where settings are changed and it's verified that they're still changed on new app version - one run for "use custom DNS" setting and one run for all other settings.

All tests are scheduled to run nightly(`ios-end-to-end-tests.yml` and `ios-end-to-end-tests-settings-migration.yml`)

The easiest way to test these changes is to trigger the workflow for settings migration(`ios-end-to-end-tests-settings-migration.yml`) on my mullvadvpn-app fork which will execute the tests on my computer and phone, but requires some coordination. The tests themselves can be tested locally by:
1. Check out commit `db8e804996fbbbd6fe500e62785a7e81c8a68a5c`
2. Run `MullvadVPNUITestsChangeDNSSettings` test plan
3. Check out latest commit in PR branch
4. Run `MullvadVPNUITestsVerifyDNSSettingsChanged` test plan
5. Check out commit `db8e804996fbbbd6fe500e62785a7e81c8a68a5c`
6. Run `MullvadVPNUITestsChangeSettings `
7. Check out latest commit in PR branch
8. Run `MullvadVPNUITestsVerifySettingsChanged` test plan

Note that if you want to run the tests locally there's a new configuration parameter in `UITests.xcconfig` which needs to be added:
```
SHOULD_BE_REACHABLE_DOMAIN = mullvad.net
```




- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6022)
